### PR TITLE
Updated mac install guide and sklearn svm tutorial

### PR DESCRIPTION
**Pull request by MarkSong535**
This PR compose of two main parts: Mac install guide update and SKLearn svm tutorial.

Mac install guide updates overview:
- Add (or update) deprecation indicator to docs that should be deprecated (M-series install guides).
- Fix quick start guide that categorize intel based Mac same as linux.
- Add translated Mac install guide
- Add Mac install guide that is universally applicable for both intel and arm based Macs.

SKLearn svm tutorial addition overview:
- Added bilingual SKLearn svm tutorial.

### CODE REVIEW

File: docs/source/contribs/development_Mac.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+For Tensorflow version 2.13 or later:
+```bash
+pip install tensorflow
+pip install tensorflow-metal
+```
+
+For Tensorflow version 2.12 or earlier:
+```bash
+pip install tensorflow-macos
```
Comment (refraction-ray) [inline docs/source/contribs/development_Mac.md:52]: is this work for intel mac?

File: docs/source/contribs/development_Mac.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+model = tf.keras.applications.ResNet50(
+    include_top=True,
+    weights=None,
+    input_shape=(32, 32, 3),
+    classes=100,)
+
+loss_fn = tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True)
+model.compile(optimizer="adam", loss=loss_fn, metrics=["accuracy"])
+model.fit(x_train, y_train, epochs=5, batch_size=64)
+```
```
Comment (refraction-ray) [inline docs/source/contribs/development_Mac.md:72]: I suggest an verify example using ``tc.quantum.heisenberg_hamiltonian``, as this one uses the weakest support part of tf

Comment (refraction-ray) [reply docs/source/contribs/development_Mac.md:72]: and I suggest you to test to ensure that jax and tf can work together in mac m1

File: docs/source/quickstart.rst [hunk 3]
```diff
@@ -29,7 +29,7 @@ For more details on docker setup, please refer to `docker readme <https://github
 
 - For Windows, due to the lack of support for Jax, we recommend to use docker or WSL, please refer to `TC via windows docker <contribs/development_windows.html>`_ or `TC via WSL <contribs/development_wsl2.html>`_.
 
-- For Mac with M series chips (arm architecture), please refer to `TC on Mac M series <contribs/development_MacARM.html>`_.
+- For MacOS, please refer to `TC on Mac <contribs/development_Mac.html>`_ or `在Mac上安装TC <contribs/development_Mac_cn.html>`_.
```
Comment (refraction-ray) [inline docs/source/quickstart.rst:32]: dont add chinese translation link here.

File: docs/source/quickstart.rst [hunk 4]
```diff
@@ -5,7 +5,7 @@ Quick Start
 Installation
 --------------
 
-- For x86 Linux or Mac, 
+- For x86 Linux,
```
Comment (refraction-ray) [inline docs/source/quickstart.rst:8]: I suggest to keep intel mac here, so that the user dont need to refer to further docs

File: docs/source/tutorial.rst [hunk 5]
```diff
@@ -24,4 +24,5 @@ Jupyter Tutorials
     tutorials/barren_plateaus.ipynb
     tutorials/qaoa_portfolio_optimization.ipynb
     tutorials/imag_time_evo.ipynb
-    tutorials/qcloud_sdk_demo.ipynb
\ No newline at end of file
+    tutorials/qcloud_sdk_demo.ipynb
+    tutorials/sklearn_svc.ipynb
```
Comment (refraction-ray) [inline docs/source/tutorial.rst:28]: please move this one row higher: i.e. keep qcloud demo as the last tutorial

File: docs/source/tutorials/sklearn_svc.ipynb [hunk 6]
```diff
@@ -0,0 +1,303 @@
+{
```
Comment (refraction-ray) [inline docs/source/tutorials/sklearn_svc.ipynb:1]: <p>we use the SKLearn library to implement SVC in the following</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tencent-quantum-lab/tensorcircuit/pull/164/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (refraction-ray) [inline docs/source/tutorials/sklearn_svc.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #2.        f = open("german.data-numeric")
</pre><p>where is the data file? I suggest that in the tutorial, we can obtain the data automatically via API or from the web</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tencent-quantum-lab/tensorcircuit/pull/164/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (refraction-ray) [inline docs/source/tutorials/sklearn_svc.ipynb:1]: <p>better visualize the circuit using c.draw()</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tencent-quantum-lab/tensorcircuit/pull/164/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Comment (refraction-ray) [inline docs/source/tutorials/sklearn_svc.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #17.            data_ret = K.cast(K.power(x_qt @ K.transpose(y_qt), 2), "float32")
</pre><p>it is not the square of the inner product, it should be the square norm of the inner product $|\langle x\vert \rangle |^2$</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tencent-quantum-lab/tensorcircuit/pull/164/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

File: docs/source/contribs/development_Mac.md (hunk 1 above)
Comment (MarkSong535) [reply docs/source/contribs/development_Mac.md:52]: Yes

File: docs/source/quickstart.rst (hunk 3 above)
Comment (MarkSong535) [reply docs/source/quickstart.rst:32]: Where should I add it?

File: docs/source/tutorials/sklearn_svc.ipynb (hunk 6 above)
Comment (MarkSong535) [reply docs/source/tutorials/sklearn_svc.ipynb:1]: Due to unknown error, c.draw() cannot be added under the function where the circuit is created.

File: docs/source/quickstart.rst (hunk 3 above)
Comment (refraction-ray) [reply docs/source/quickstart.rst:32]: > Where should I add it?

maybe in your mac english installlation doc

``for chinese version of the doc, see link``

File: docs/source/contribs/development_Mac.md (hunk 2 above)
Comment (refraction-ray) [reply docs/source/contribs/development_Mac.md:72]: > Both were tested and both (`tc.quantum.heisenberg_hamiltonian` and Jax with tf) can operate normally.

great!

File: docs/source/contribs/development_Mac.md (hunk 1 above)
Comment (refraction-ray) [reply docs/source/contribs/development_Mac.md:52]: are you indicating that there is a package tensorflow-macos also on intel mac？ I do think install tensorflow is the best solution for intel mac for these early versions

File: docs/source/tutorials/sklearn_svc.ipynb (hunk 6 above)
Comment (MarkSong535) [reply docs/source/tutorials/sklearn_svc.ipynb:1]: fixed!

Comment (MarkSong535) [reply docs/source/tutorials/sklearn_svc.ipynb:1]: fixed!

Comment (MarkSong535) [reply docs/source/tutorials/sklearn_svc.ipynb:1]: fixed!

File: docs/source/tutorial.rst (hunk 5 above)
Comment (MarkSong535) [reply docs/source/tutorial.rst]: fixed!

File: docs/source/quickstart.rst (hunk 4 above)
Comment (MarkSong535) [reply docs/source/quickstart.rst:8]: This is not specifically the gpu version of tensorflow on MacOS. That version is tensorflow-metal and not installed in this tutorial and it is indeed slower than apple optimized version of tensorflow.
A benchmark is underway.

File: docs/source/quickstart.rst (hunk 3 above)
Comment (MarkSong535) [reply docs/source/quickstart.rst]: fixed.

Added it to translation file.

File: docs/source/contribs/development_Mac.md (hunk 1 above)
Comment (MarkSong535) [reply docs/source/contribs/development_Mac.md:52]: It should be updated and remove install guide for tensorflow version 2.12 or earlier. For tf of those versions, tf-macos and vanilla tf are both not functioning as desired on tested platforms.

Comment (MarkSong535) [reply docs/source/contribs/development_Mac.md]: updated and please refer to the latest updated version as this issues have been resolved in latest update.

File: docs/source/quickstart.rst (hunk 4 above)
Comment (MarkSong535) [reply docs/source/quickstart.rst:8]: A benchmark result have been attached to the end of the latest version of the install guide

File: docs/source/contribs/development_Mac.md [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+    <td>0.046412581s</td>
+  </tr>
+  <tr>
+    <td>Total time</td>
+    <td>11.72371747s</td>
+    <td>11.55208979s</td>
+    <td>16.25165417s</td>
+  </tr>
+</table>
+
```
Comment (refraction-ray) [inline docs/source/contribs/development_Mac.md:111]: could you also provde info on what specifically called Vanilla/Apple tensorflow? i.e. how can we install each respectively

Comment (refraction-ray) [reply docs/source/contribs/development_Mac.md:111]: and seems to me that vanilla tensorflow is the same as mac tensorflow, then could we revert the line in quick start, where intel mac users is also directed to the simple pip install?

**Comment by refraction-ray** [comment]
for the german data part, could we present a way that can obtain the data without uploading the data directly on the codebase? say using sklearn/tf api or directly download from somewhere on the internet

File: docs/source/contribs/development_Mac.md (hunk 7 above)
Comment (MarkSong535) [reply docs/source/contribs/development_Mac.md:111]: > could you also provde info on what specifically called Vanilla/Apple tensorflow? i.e. how can we install each respectively

This is explained during tf install guide.

Comment (MarkSong535) [reply docs/source/contribs/development_Mac.md:111]: > could you also provde info on what specifically called Vanilla/Apple tensorflow? i.e. how can we install each respectively

This is explained in the install tf part. Vanilla tf is described as the version that Google provides.

Comment (MarkSong535) [reply docs/source/contribs/development_Mac.md:111]: > and seems to me that vanilla tensorflow is the same as mac tensorflow, then could we revert the line in quick start, where intel mac users is also directed to the simple pip install?

They can, however, if installed without considering whether tf-deps is installed, they will have not method of controlling which version of tf in specific that they are installing. It is, based on my own pov, beneficial to be put in here so people can understand and gain control of which version of tf that they are going to install.

File: docs/source/tutorials/sklearn_svc.ipynb (hunk 6 above)
Comment (refraction-ray) [reply docs/source/tutorials/sklearn_svc.ipynb:1]: it is ok now since the circuit structure is more clearly organized

File: docs/source/contribs/development_Mac.md (hunk 7 above)
Comment (refraction-ray) [reply docs/source/contribs/development_Mac.md:111]: ok, cool. it is okay to me to keep the doc this way :)

**Comment by MarkSong535** [comment]
> for the german data part, could we present a way that can obtain the data without uploading the data directly on the codebase? say using sklearn/tf api or directly download from somewhere on the internet

Fixed!

**Review by refraction-ray — APPROVED** [review]
LGTM, thanks for the contribution, especially the extensive exploration on the installation subtlety on macOS
